cmd/generate/config/rules/gcp.go

```diff
@@ -29,8 +29,8 @@ func GCPAPIKey() *config.Rule {
 	r := config.Rule{
 		RuleID:      "gcp-api-key",
 		Description: "Uncovered a GCP API key, which could lead to unauthorized access to Google Cloud services and data breaches.",
-		Regex:       generateUniqueTokenRegex(`AIza[0-9A-Za-z\\-_]{35}`, true),
-
+		Regex:       generateUniqueTokenRegex(`(?-i:AIza)[0-9a-z_-]{35}`, true),
```

Comment:
Fun fact: `\\-_` does not match `-` and `_`, it matches characters between `\` and `_`.
![image](https://github.com/user-attachments/assets/7f41f235-b2d0-47a0-b6a1-351fdb942ce8)